deploy/csv/marketplace.csv.yaml

```diff
@@ -0,0 +1,103 @@
+    The operator will then create a CatalogSourceConfig CR which will, for the time being, trigger the marketplace operator to create a ConfigMap CR and CatalogSource CR. The package-server, managed by [OLM](https://github.com/operator-framework/operator-lifecycle-manager), will then respond to the creation of these CRs and allow the external operators to be visible in the [marketplace UI](https://github.com/openshift/console/tree/master/frontend/public/components/marketplace).
+  keywords: ['marketplace', 'catalog', 'olm', 'admin']
+  version: 0.0.1
+  maturity: alpha
+  maintainers:
+  - name: AOS Marketplace Team
+    email: aos-marketplace@redhat.com
+  provider:
+    name: Red Hat
+  labels:
+    name: marketplace-operator
+  selector:
+    matchLabels:
+      name: marketplace-operator
+  links:
+  - name: Markplace Operator Source Code
+    url: https://github.com/operator-framework/operator-marketplace
+  install:
+    strategy: deployment
+    spec:
+      clusterPermissions:
+      - serviceAccountName: default
+        rules:
+        - apiGroups:
+          - marketplace.redhat.com
+          resources:
+          - "*"
+          verbs:
+          - "*"
+        - apiGroups:
+          - ""
+          resources:
+          - services
+          - configmaps
+          verbs:
+          - "*"
+        - apiGroups:
+          - operators.coreos.com
+          resources:
+          - catalogsources
+          verbs:
+          - "*"
+      deployments:
+      - name: marketplace-operator
+        spec:
+          replicas: 1
+          selector:
+            matchLabels:
+              name: marketplace-operator
+          template:
+            metadata:
+              name: marketplace-operator
+              labels:
+                name: marketplace-operator
+            spec:
+              containers:
+                - name: marketplace-operator
+                  image: quay.io/redhat/marketplace
+                  ports:
+                  - containerPort: 60000
+                    name: metrics
+                  command:
+                  - marketplace-operator
+                  imagePullPolicy: Always
+                  env:
+                    - name: WATCH_NAMESPACE
+                      valueFrom:
+                        fieldRef:
+                          fieldPath: metadata.namespace
+                    - name: OPERATOR_NAME
+                      value: "marketplace-operator"
+customresourcedefinitions:
```

Comment:
Please look in to [x-descriptors](https://github.com/openshift/console/blob/master/frontend/public/components/operator-lifecycle-manager/descriptors/types.ts#L5-L26)

Reply:
I looked into `x-descriptors` as you asked.

`x-descriptors` are a part of the `specDescriptor`, which is optional. Right now, the primary use case of a `specDescriptor` is to get UI details to the console, but a `specDescriptor` is intended to be general semantic details about the CRD fields which could link Operators together. For example, an `exposedDatabaseEndpoint` output capability on one Operator  could be matched with a `databaseEndpoint` input on another.

We had spoken briefly about the `specDescriptor` last week and I believe that we decided not to include them since Marketplace wouldn't show up in the UI. If we'd like to include them  please let me know.

Reply:
I am not sure how `Marketplace` would not show up in the OLM UI as it needs to be in a `CatalogSource` in the OLM namespace or `OperatorGroup` for subscriptions to work.

Reply:
I see your point, I'll be sure to add the specDescriptors.

Reply:
I have added `specDescriptors` and `statusDescriptors` in the most recent commit.

The was a single [x-descriptor](https://github.com/openshift/console/blob/master/frontend/public/components/operator-lifecycle-manager/descriptors/types.ts#L5-L26) that matched a field defined in the `Marketplace Operator` status/spec - no other `x-descriptor` seemed to fit. (In this case the  `k8sPhase x-descriptor` matched well with the `currentPhase` status).

If we want to extend the existing x-descriptor we can do so by following the directions shown [here](https://github.com/openshift/console/tree/master/frontend/public/components/operator-lifecycle-manager/descriptors).